### PR TITLE
CRIBL-25846 

### PR DESCRIPTION
**Pull request by fbonecco-cribl**
Find RC analysis in this comment: https://taktak.atlassian.net/browse/CRIBL-25846?focusedCommentId=554912.

### Test suite

Run the whole test suite and make sure it passes. Most of these tests are purely E2E and rely on a real Kafka cluster that gets spun up as part of the `yarn test` command. That said...the suite is really flakey and it might take you a couple of runs to  have them all passing (if you're lucky). Double check the flakiness is also present on the `cribl-release` and/or `master` branches.

1. Check out this repo and branch
2. `npm install --global yarn`
3. `yarn install`
4. `yarn test`

Or you can run just the `network` suite, where changes in this PR would land to:
`npm run test:group:network`

### Manual testing in Stream

Find the instructions in this PR https://bitbucket.org/cribl/cribl/pull-requests/21015 to test this patched version with real Stream.

### CODE REVIEW

File: src/network/connection.spec.js [hunk 1]
```diff
@@ -98,6 +98,62 @@ describe('Network > Connection', () => {
       expect(connection.socket.end).toHaveBeenCalled()
       expect(connection.socket.unref).toHaveBeenCalled()
     })
+
```
Comment (jspitzig-cribl) [inline src/network/connection.spec.js:101]: @fbonecco-cribl This gives me an error about "unnecessary escape" when I try to run the tests.

File: package.json [hunk 2]
```diff
@@ -35,7 +35,8 @@
     "test:group:admin": "yarn jest --forceExit --testPathPattern 'src/admin/.*'",
     "test:group:producer": "yarn jest --forceExit --testPathPattern 'src/producer/.*'",
     "test:group:consumer": "yarn jest --forceExit --testPathPattern 'src/consumer/.*.spec.js'",
```
Comment (jspitzig-cribl) [inline package.json:37]: @fbonecco-cribl so, were the network tests just not being run, then? :disappointed:

File: src/network/connection.spec.js (hunk 1 above)
Comment (fbonecco-cribl) [reply src/network/connection.spec.js:101]: Thanks, should be fixed now.

File: package.json (hunk 2 above)
Comment (fbonecco-cribl) [reply package.json:37]: `test:group:network` is a just a new alias to be able to run these network-related tests alone. These are also executed as part of the main suite.

Comment (jspitzig-cribl) [reply package.json:37]: ah, gotcha. i was basing it on the fact that you also added to a larger list below, but i just realized that's also a group, and they should be run in the general 'test' command.

File: src/network/connection.js [hunk 3]
```diff
@@ -271,6 +271,11 @@ module.exports = class Connection {
     await this.requestQueue.waitForPendingRequests()
     this.requestQueue.destroy()
 
+    this.bytesNeeded = Decoder.int32Size()
```
Comment (brianfranko22) [inline src/network/connection.js:274]: Can we just add a quick comment explaining why we are doing this here?

**Review by brianfranko22 — APPROVED** [review]

File: src/network/connection.js (hunk 3 above)
Comment (fbonecco-cribl) [reply src/network/connection.js:274]: Do you mean around `#274` in particular?

**Comment by awhittier-cribl** [comment]
Changes make sense to me. Network tests pass with the change and fail without it. The full test battery passes (eventually) except for "Protocol > Requests > Produce > v2 › request with gzip" which fails for me on the cribl-release branch as well.

**Review by awhittier-cribl — APPROVED** [review]
Changes make sense to me. Network tests pass with the change and fail without it. The full test battery passes (eventually) except for "Protocol > Requests > Produce > v2 › request with gzip" which fails for me on the cribl-release branch as well.

File: src/network/connection.js (hunk 3 above)
Comment (brianfranko22) [reply src/network/connection.js:274]: These 4 lines that were added.

**Comment by jspitzig-cribl** [comment]
Tested manually and ran UTs. Everything is working for me. Looks good! :+1:

**Review by jspitzig-cribl — APPROVED** [review]

**Pull request merged**